distributed/http/scheduler/api.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import json
+
+from distributed.http.utils import RequestHandler
+
+
+class APIHandler(RequestHandler):
+    def get(self):
+        self.write("API V1")
+        self.set_header("Content-Type", "text/plain")
+
+
+class RetireWorkersHandler(RequestHandler):
+    async def post(self):
+        self.set_header("Content-Type", "text/json")
+        scheduler = self.server
+        try:
+            params = json.loads(self.request.body)
+            workers_info = await scheduler.retire_workers(**params)
+            self.write(json.dumps(workers_info))
+        except Exception as e:
+            self.set_status(400, str(e))
+            self.write(json.dumps({"Error": "Bad request"}))
+
+
+class WorkersToCloseHandler(RequestHandler):
+    async def post(self):
+        self.set_header("Content-Type", "text/json")
+        scheduler = self.server
+        try:
+            params = json.loads(self.request.body)
+            workers_to_close = {"workers": scheduler.workers_to_close(**params)}
+            self.write(json.dumps(workers_to_close))
+        except Exception as e:
+            self.set_status(400, str(e))
```

Comment:
a generic exception is not a 400 but rather a 500. A 400 would indicate wrong parameters, etc. but a generic exception might be anything (e.g. serialization error of the workers_to_close dictionary)

Reply:
Agree this should be 500.